### PR TITLE
Document Iceberg connector supported data types mapping 

### PR DESCRIPTION
**Pull request by sbabtiwa**
## Description
Added documentation for supported Iceberg connector data types mapping.

## Motivation and Context

This change is required to help users understand which PrestoDB and Iceberg data types successfully map to each other. 

This issue fixes [21848.](https://github.com/prestodb/presto/issues/21848)

## Impact

Adding documentation to Presto Docs iceberg.rst file. 

## Test Plan

Tested changes by building and viewing the Presto documentation. 

## Contributor checklist

- [X] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [X] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [X] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

General Changes
* Add documentation for supported data types mapping in the Iceberg connector.

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 1]
```diff
@@ -1136,3 +1136,98 @@ In the following query, the expression CURRENT_TIMESTAMP returns the current tim
...
+  * - ``INTEGER``
+    - ``INTEGER``
+  * - ``TIME``
+    - ``TIME``
+  * - ``TIMESTAMP``
+    - ``TIMESTAMP``
+  * - ``STRING``
+    - ``VARCHAR``
+  * - ``LIST``
+    - ``LIST``
```
Comment (hantangwangd) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:1182]: Presto type should be `ARRAY`?

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -1136,3 +1136,98 @@ In the following query, the expression CURRENT_TIMESTAMP returns the current tim
...
+  * - ``VARCHAR``
+    - ``STRING``
+  * - ``VARBINARY``
+    - ``BINARY``
+  * - ``DATE``
+    - ``DATE``
+  * - ``ROW``
+    - ``ROW``
+  * - ``ARRAY``
+    - ``ARRAY``
```
Comment (hantangwangd) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:1223]: Iceberg type should be `LIST`?

**Review by hantangwangd — COMMENTED** [review]
Thanks for the document supplement. Some little nits.

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 3]
```diff
@@ -1136,3 +1136,98 @@ In the following query, the expression CURRENT_TIMESTAMP returns the current tim
...
+  * - ``DECIMAL``
+    - ``DECIMAL``
+  * - ``VARCHAR``
+    - ``STRING``
+  * - ``VARBINARY``
+    - ``BINARY``
+  * - ``DATE``
+    - ``DATE``
+  * - ``ROW``
+    - ``ROW``
```
Comment (hantangwangd) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:1221]: STRUCT?

**Comment by steveburnett** [comment]
Thanks for this! A local build of the docs looks good in terms of formatting and general docs concerns. I'll wait to review and approve until after @hantangwangd's comments are addressed. Tag me in a comment or request a review from me when this is ready!

**Comment by sbabtiwa** [comment]
Hi @hantangwangd, @steveburnett, and @tdcmeehan, 

I have modified the Iceberg connector type mapping documentation to implement the comments above and make the content more original. Please let me know if additional changes are needed. 

Thanks

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 4]
```diff
@@ -1179,11 +1179,9 @@ The connector maps Iceberg types to the corresponding PrestoDB types:
   * - ``STRING``
     - ``VARCHAR``
   * - ``LIST``
-    - ``LIST``
+    - ``ARRAY``
   * - ``MAP``
     - ``MAP``
-  * - ``STRUCT``
-    - ``ROW``
```
Comment (hantangwangd) [inline presto-docs/src/main/sphinx/connector/iceberg.rst]: Why remove iceberg ``STRUCT`` to presto ``ROW`` mapping?

File: presto-docs/src/main/sphinx/connector/iceberg.rst (hunk 3 above)
Comment (hantangwangd) [reply presto-docs/src/main/sphinx/connector/iceberg.rst:1219]: Here, presto ``ROW`` should be mapped to iceberg ``STRUCT``?

**Review by hantangwangd — COMMENTED** [review]
Still little nits

File: presto-docs/src/main/sphinx/connector/iceberg.rst (hunk 3 above)
Comment (sbabtiwa) [reply presto-docs/src/main/sphinx/connector/iceberg.rst:1219]: I thought last week's comment meant I should remove the STRUCT type mapping, but I see from today's comment that I should modify the ROW/ROW PrestoDB to Iceberg mapping to ROW/STRUCT

File: presto-docs/src/main/sphinx/connector/iceberg.rst (hunk 4 above)
Comment (sbabtiwa) [reply presto-docs/src/main/sphinx/connector/iceberg.rst]: Please see comment above: https://github.com/prestodb/presto/pull/21998#discussion_r1509248887. I will add it back.

**Comment by sbabtiwa** [comment]
@hantangwangd, I have modified the STRUCT/ROW and vice versa type mapping in the branch.

**Review by steveburnett — DISMISSED** [review]
LGTM! (docs)

Pull updated branch, new local build. Thanks!

File: presto-docs/src/main/sphinx/connector/iceberg.rst (hunk 3 above)
Comment (hantangwangd) [reply presto-docs/src/main/sphinx/connector/iceberg.rst]: Oh, sorry for not saying clearly.

**Review by hantangwangd — APPROVED** [review]
LGTM! Thanks for the fix!

**Comment by sbabtiwa** [comment]
@tdcmeehan, I ran into issues while squashing commits in my original branch, so I moved the single commit into a recreated branch. As a result, I had to open a new PR: https://github.com/prestodb/presto/pull/22093. Sorry for the inconvenience.

**Pull request closed without merging**